process/block/baseProcess.go

```diff
@@ -3304,3 +3304,85 @@ func (bp *baseProcessor) getCurrentBlockHeader() data.HeaderHandler {
 
 	return bp.blockChain.GetGenesisHeader()
 }
+
+func (bp *baseProcessor) checkContextBeforeExecution(header data.HeaderHandler) error {
+	lastCommittedRootHash, err := bp.accountsDB[state.UserAccountsState].RootHash()
+	if err != nil {
+		return err
+	}
+
+	// TODO: the GetLastExecutedBlockInfo should return also the LastCommittedBlockInfo (in case the committed block was V2)
+	// this is done on another PR
+	lastExecutedNonce, lastExecutedHash, lastExecutedRootHash := bp.blockChain.GetLastExecutedBlockInfo()
+	if !bytes.Equal(header.GetPrevHash(), lastExecutedHash) {
+		return process.ErrBlockHashDoesNotMatch
+	}
+	if header.GetNonce() != lastExecutedNonce+1 {
+		return process.ErrWrongNonceInBlock
+	}
+	if !bytes.Equal(lastCommittedRootHash, lastExecutedRootHash) {
```

Comment:
[question]
Is it mandatory that the last committed is the last executed?
Can't you have multiple executed headers on top of last committed?

Or is this related to the `TODO` above?